### DISCUSSION TITLE
How to compile with mpeg support?

### DISCUSSION THREAD

**Discussion by SoftCreatR** [ImageMagick/ImageMagick#3530] [Help] [answered]
Hi,

according to [this list](https://imagemagick.org/script/formats.php), ImageMagick should have mpeg delegate support, if ffmpeg is installed. However, there's no `--with-mpeg` or `--with-mpeg2` option (but there was in older version, if you can believe Google). So, if mpeg support still exists in ImageMagick, how can I activate it?

EDIT:

`$ identify -list format` lists MPEG with rw+, so does that mean, that support exists, it's just not listed in the list of delegates?

**Comment by snibgo** [comment, marked as answer]
--with-ABC is for installing _internal_ delegates, with libraries, as listed in the "Delegates (built-in)" section of "magick -version". But ffmpeg is used as an _external_ delegate, by calling the binary program. We can see this by converting with "-verbose":
```
f:\web\im>%IMG7%magick -verbose xc: m.mp4

xc:=> XC 1x1 1x1+0+0 16-bit sRGB 0.000u 0:00.000
xc:=>C:/Users/Alan/AppData/Local/Temp/magick-2236ZGfOxanK9NLz0.jpg XC 1x1 1x1+0+
0 16-bit Bilevel Gray 160B 0.000u 0:00.034
"ffmpeg.exe" -nostdin -v -1 -i "C:/Users/Alan/AppData/Local/Temp/magick-2236ZGfO
xanK9NLz%d.jpg" "C:/Users/Alan/AppData/Local/Temp/magick-22364xKGkpCgePNi.mp4"
xc:=>m.mp4 XC 1x1 1x1+0+0 16-bit sRGB 0.063u 0:01.702
```

**Reply by SoftCreatR** [reply]
So having ffmpeg installed is enough and IM "magically" provides support for mpeg & co?

**Comment by snibgo** [comment]
Yes. On my Windows 8.1 laptop, ffmpeg isn't on the system path. But it is installed in the same directory as the IM binaries, as part of the IM package.

(ffmpeg is simply a binary program, a single file.)

If you have IM, I suggest you try the command in my previous reply:
```
magick -verbose xc: m.mp4
```
